glusterd: use binary uuid to avoid extra conversions

dmantipov opened the pull request:
Prefer binary representation of UUID in peer, friend and snapshot
management code, which is expected to simplify the things and avoid
extra UUID conversions between text and binary representations
within `glusterd`. This comes with a presumably less critical overhead
of doing a few more conversions on client side, which is needed for
printing different objects' status in both textual and XML form.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/xlators/mgmt/glusterd/src/glusterd-op-sm.c b/xlators/mgmt/glusterd/src/glusterd-op-sm.c
index 918e45de7..1d9ae76f7 100644
--- a/xlators/mgmt/glusterd/src/glusterd-op-sm.c
+++ b/xlators/mgmt/glusterd/src/glusterd-op-sm.c
@@ -4903,8 +4903,7 @@ glusterd_op_volume_dict_uuid_to_hostname(dict_t *dict, const char *key_fmt,
 
         hostname = glusterd_uuid_to_hostname(uuid);
         if (hostname) {
-            gf_msg_debug(this->name, 0, "%s -> %s",
-                         uuid_utoa(uuid), hostname);
+            gf_msg_debug(this->name, 0, "%s -> %s", uuid_utoa(uuid), hostname);
             ret = dict_set_dynstrn(dict, key, keylen, hostname);
             if (ret) {
                 gf_msg(this->name, GF_LOG_ERROR, 0, GD_MSG_DICT_SET_FAILED,
@@ -5244,7 +5243,7 @@ glusterd_op_modify_op_ctx(glusterd_op_t op, void *ctx)
 
             /* Note 'uuid' will be used to convert to hostname later. */
             {
-                uuid_t uuid = { 0 };
+                uuid_t uuid = {0};
                 int i;
 
                 for (i = 1; i <= count; i++) {
@@ -5252,7 +5251,7 @@ glusterd_op_modify_op_ctx(glusterd_op_t op, void *ctx)
                     ret = dict_get_gfuuid(op_ctx, key, &uuid);
                     if (!ret) {
                         snprintf(key, sizeof(key), "node-name-%d", i);
-                        ret  = dict_set_gfuuid(op_ctx, key, uuid, true);
+                        ret = dict_set_gfuuid(op_ctx, key, uuid, true);
                         if (ret != 0) {
                             gf_msg_debug(this->name, 0,
                                          "failed to set node name");
```

dmantipov commented:
/run regression

dmantipov commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

3 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/2385/

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/glusterfs/glusterfs.h b/libglusterfs/src/glusterfs/glusterfs.h
index 12ab40da9..4ed992f36 100644
--- a/libglusterfs/src/glusterfs/glusterfs.h
+++ b/libglusterfs/src/glusterfs/glusterfs.h
@@ -63,7 +63,6 @@
 #define O_PATH 010000000 /* from asm-generic/fcntl.h */
 #endif
 
-
 #ifndef EBADFD
 /* Mac OS X does not have EBADFD */
 #define EBADFD EBADF
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/glusterfs/glusterfs.h b/libglusterfs/src/glusterfs/glusterfs.h
index 12ab40da9..4ed992f36 100644
--- a/libglusterfs/src/glusterfs/glusterfs.h
+++ b/libglusterfs/src/glusterfs/glusterfs.h
@@ -63,7 +63,6 @@
 #define O_PATH 010000000 /* from asm-generic/fcntl.h */
 #endif
 
-
 #ifndef EBADFD
 /* Mac OS X does not have EBADFD */
 #define EBADFD EBADF
```

dmantipov commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

0 test(s) generated core 


6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

5 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/2472/

Commit:
glusterd, cli: prefer binary uuid to avoid extra conversions

Prefer binary representation of UUID in peer, friend and snapshot
management code, which is expected to simplify the things and avoid
extra UUID conversions between text and binary representations
within glusterd. This comes with a presumably less critical overhead
of doing a few more conversions on client side, which is needed for
printing different objects' status in both textual and XML form.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

dmantipov commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

0 test(s) generated core 


6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

5 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/2488/

dmantipov commented:
/run regression

pranithk commented:
I think this breaks protocol. Older version glusterds will expect textual input I think.

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

0 test(s) generated core 


6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

5 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/2500/

dmantipov commented:
Closed.
